src/components/TruncatedQuery/TopQueriesTruncatedQuery.tsx

```diff
@@ -0,0 +1,20 @@
+import cn from 'bem-cn-lite';
+
+import {CellWithPopover} from '../CellWithPopover/CellWithPopover';
+
+import './TruncatedQuery.scss';
+
+const b = cn('kv-truncated-query');
+
+interface TopQueriesTruncatedQueryProps {
+    value: string | undefined;
+}
+
+export const TopQueriesTruncatedQuery = ({value = ''}: TopQueriesTruncatedQueryProps) => {
+    const content = value.split('\n').slice(0, 5).join('\n');
+    return (
+        <CellWithPopover contentClassName={b('popover-content')} content={content}>
+            {value}
+        </CellWithPopover>
+    );
+};
```

Comment:
Why only 5 lines is displayed in `Popover`?

Regarding `TruncatedQuery` and `TopQueriesTruncatedQuery`. I think they are two realisations of the same component, you could place them in 1 file. And you can call it something like `TruncatedQueryWithPopover`. Current naming is not good, because you have `TopQueries` in `Diagnostics`, but don't use your component in there. Second component should either be able to display data with defined `maxQueryHeight` (where 1 is special case with `white-space: nowrap;`) or have comment that it's one line query display. Or maybe it could even be named `OneLineQueryWithPopover`

Generally, it's good to name component scope-based, when they are part of some larger components and have defined place in component tree. And in this case they are placed near parent components. But for the more general-purpose components (we have them in `src/components` folder) it better to chose names, that reflects components purpose and functionality